kevinene91/Store-Manager-API
# #161378934 Auth Endpoints

# What does this PR do?
It has the authentication endpoints with data persisted to a database
`POST api/v2/auth/signup`
`POST api/v2/auth/login`
`POST api/v2/auth/logout`

# Description of Task Completed?
An admin should add users 
a user should login 
a user should logout

# How to Manually test this 
- Clone the repository `git clone https://github.com/kevinene91/Store-Manager-API.git`
- Change into the app directory `cd Store-Manager-API`
- get this branch code `git fetch origin ft-database-auth-161378934`
- Navigate the folder structure
- create a virtual environment `python3 -m pyvenv venv`
- pip install -r requirements.txt
Export environment variables
     - export SECRET_KEY ="random string here"
    - export APP_SETTINGS="development"
   - export DATABASE_URL="dbname=storemanager user= password= host=localhost"
-  navigate to the folder `app/test/v2`
- run pytest test_auth.py tests should pass
navigate back to root and run python run.py 
use postman to test the endpoints 
### Relevant PVT story? 

[161378934](https://www.pivotaltracker.com/n/projects/2203367/stories/161378934)
